x/auth/tx/config/config.go

```diff
@@ -58,22 +55,26 @@ type ModuleOutputs struct {
 	BaseAppOption runtime.BaseAppOption
 }
 
-// ProvideSignModeOptions provides the default x/tx SignModeOptions for the SDK.
-func ProvideSignModeOptions(bk BankKeeper) tx.SignModeOptions {
-	opts, err := NewSignModeOptionsWithMetadataQueryFn(NewBankKeeperCoinMetadataQueryFn(bk))
-	if err != nil {
-		panic(err)
-	}
-	return opts
+func ProvideProtoRegistry() txsigning.ProtoFileResolver {
+	return registry.MergedProtoRegistry()
 }
 
 func ProvideModule(in ModuleInputs) ModuleOutputs {
-	var txConfig client.TxConfig
 	var customSignModeHandlers []txsigning.SignModeHandler
 	if in.CustomSignModeHandlers != nil {
 		customSignModeHandlers = in.CustomSignModeHandlers()
 	}
-	txConfig = tx.NewTxConfigWithOptions(in.ProtoCodecMarshaler, in.SignModeOptions, customSignModeHandlers...)
+	sdkConfig := sdk.GetConfig()
+	txConfigOptions := tx.ConfigOptions{
+		FileResolver: in.ProtoFileResolver,
+		// From static config? But this is already in auth config.
+		// - Provide codecs there as types?
+		// - Provide static prefix there exported from config?
+		// - Just do as below?
+		AddressCodec:   authcodec.NewBech32Codec(sdkConfig.GetBech32AccountAddrPrefix()),
+		ValidatorCodec: authcodec.NewBech32Codec(sdkConfig.GetBech32ValidatorAddrPrefix()),
```

Comment:
I'd like some feedback re the above comments. What approach do you think is best?

Reply:
We will need to find an alternative way to do this. I think it's fine for now as I did something similar, but I intend to address in a follow up.